### PR TITLE
check presence of tekton crds before initializing controller manager

### PR DESCRIPTION
**Pull request by gabemontero**
After analyzing results from https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/redhat-appstudio_jvm-build-service/144/pull-ci-redhat-appstudio-jvm-build-service-main-jvm-build-service-in-repo-e2e/1553029354840133632 and in particluar https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/redhat-appstudio_jvm-build-service/144/pull-ci-redhat-appstudio-jvm-build-service-main-jvm-build-service-in-repo-e2e/1553029354840133632/artifacts/jvm-build-service-in-repo-e2e/gather-extra/artifacts/pods/build-service_build-service-controller-manager-6b4d9565b8-wthpm_manager.log 

```
I0729 15:04:05.852597       1 request.go:601] Waited for 1.019344803s due to client-side throttling, not priority and fairness, request: GET:https://172.30.0.1:443/apis/samples.operator.openshift.io/v1?timeout=32s
1.659107047555279e+09	INFO	controller-runtime.metrics	Metrics server is starting to listen	{"addr": "127.0.0.1:8080"}
1.6591070502596347e+09	INFO	setup	starting manager
1.6591070502599032e+09	INFO	Starting server	{"path": "/metrics", "kind": "metrics", "addr": "127.0.0.1:8080"}
1.6591070502599742e+09	INFO	Starting server	{"kind": "health probe", "addr": "[::]:8081"}
I0729 15:04:10.260058       1 leaderelection.go:248] attempting to acquire leader lease build-service/5483be8f.redhat.com...
I0729 15:04:10.270017       1 leaderelection.go:258] successfully acquired lease build-service/5483be8f.redhat.com
1.6591070502700915e+09	DEBUG	events	Normal	{"object": {"kind":"Lease","namespace":"build-service","name":"5483be8f.redhat.com","uid":"58c7399c-76d7-4ccc-bc42-56456bfcfd91","apiVersion":"coordination.k8s.io/v1","resourceVersion":"42595"}, "reason": "LeaderElection", "message": "build-service-controller-manager-6b4d9565b8-wthpm_df5eec08-942f-4005-bcfa-38883a776c43 became leader"}
1.6591070502702658e+09	INFO	Starting EventSource	{"controller": "component", "controllerGroup": "appstudio.redhat.com", "controllerKind": "Component", "source": "kind source: *v1alpha1.Component"}
1.659107050270296e+09	INFO	Starting Controller	{"controller": "component", "controllerGroup": "appstudio.redhat.com", "controllerKind": "Component"}
1.6591070502703881e+09	INFO	Starting EventSource	{"controller": "pipelinerun", "controllerGroup": "tekton.dev", "controllerKind": "PipelineRun", "source": "kind source: *v1beta1.PipelineRun"}
1.659107050270438e+09	INFO	Starting Controller	{"controller": "pipelinerun", "controllerGroup": "tekton.dev", "controllerKind": "PipelineRun"}
1.6591070529735832e+09	ERROR	controller-runtime.source	if kind is a CRD, it should be installed before calling Start	{"kind": "PipelineRun.tekton.dev", "error": "no matches for kind \"PipelineRun\" in version \"tekton.dev/v1beta1\""}
sigs.k8s.io/controller-runtime/pkg/source.(*Kind).Start.func1.1
	/opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.2/pkg/source/source.go:139
k8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtectionWithContext
	/opt/app-root/src/go/pkg/mod/k8s.io/apimachinery@v0.24.2/pkg/util/wait/wait.go:233
k8s.io/apimachinery/pkg/util/wait.poll
	/opt/app-root/src/go/pkg/mod/k8s.io/apimachinery@v0.24.2/pkg/util/wait/wait.go:580
k8s.io/apimachinery/pkg/util/wait.PollImmediateUntilWithContext
	/opt/app-root/src/go/pkg/mod/k8s.io/apimachinery@v0.24.2/pkg/util/wait/wait.go:545
sigs.k8s.io/controller-runtime/pkg/source.(*Kind).Start.func1
	/opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.2/pkg/source/source.go:132
```

where that error log repeats a bunch.

Also initiated https://coreos.slack.com/archives/C03KN2B47GU/p1659116350713419

### CODE REVIEW

**Comment by gabemontero** [comment]
oops, need permission to get CRDs:

```
1.659125423419057e+09	INFO	setup	get of taskrun CRD failed with: customresourcedefinitions.apiextensions.k8s.io "taskruns.tekton.dev" is forbidden: User "system:serviceaccount:build-service:build-service-controller-manager" cannot get resource "customresourcedefinitions" in API group "apiextensions.k8s.io" at the cluster scope
```

looking to see where that is defined for build-service

**Comment by gabemontero** [comment]
So the rbac is OK now and m change seemed to help:

```
.659127885198266e+09	INFO	setup	get of taskrun CRD failed with: customresourcedefinitions.apiextensions.k8s.io "taskruns.tekton.dev" not found
1.65912789020096e+09	INFO	setup	get of taskrun CRD failed with: customresourcedefinitions.apiextensions.k8s.io "taskruns.tekton.dev" not found
1.6591278952039292e+09	INFO	setup	get of taskrun CRD failed with: customresourcedefinitions.apiextensions.k8s.io "taskruns.tekton.dev" not found
1.6591279002013462e+09	INFO	setup	get of taskrun CRD failed with: customresourcedefinitions.apiextensions.k8s.io "taskruns.tekton.dev" not found
1.6591279052017095e+09	INFO	setup	get of taskrun CRD failed with: customresourcedefinitions.apiextensions.k8s.io "taskruns.tekton.dev" not found
1.6591279102032924e+09	INFO	setup	get of taskrun CRD failed with: customresourcedefinitions.apiextensions.k8s.io "taskruns.tekton.dev" not found
1.6591279152033405e+09	INFO	setup	get of taskrun CRD failed with: customresourcedefinitions.apiextensions.k8s.io "taskruns.tekton.dev" not found
1.6591279202020762e+09	INFO	setup	get of taskrun CRD failed with: customresourcedefinitions.apiextensions.k8s.io "taskruns.tekton.dev" not found
1.6591279252138891e+09	INFO	setup	get of taskrun CRD failed with: customresourcedefinitions.apiextensions.k8s.io "taskruns.tekton.dev" not found
1.659127930213155e+09	INFO	setup	get of taskrun CRD returned successfully
I0729 20:52:11.264439       1 request.go:601] Waited for 1.045511212s due to client-side throttling, not priority and fairness, request: GET:https://172.30.0.1:443/apis/discovery.k8s.io/v1beta1?timeout=32s
1.6591279330697448e+09	INFO	controller-runtime.metrics	Metrics server is starting to listen	{"addr": "127.0.0.1:8080"}
1.6591279359414845e+09	INFO	setup	starting manager
1.6591279359576643e+09	INFO	Starting server	{"path": "/metrics", "kind": "metrics", "addr": "127.0.0.1:8080"}
1.6591279359578154e+09	INFO	Starting server	{"kind": "health probe", "addr": "[::]:8081"}
I0729 20:52:15.957920       1 leaderelection.go:248] attempting to acquire leader lease build-service/5483be8f.redhat.com...
I0729 20:52:15.978059       1 leaderelection.go:258] successfully acquired lease build-service/5483be8f.redhat.com
1.6591279359782195e+09	DEBUG	events	Normal	{"object": {"kind":"Lease","namespace":"build-service","name":"5483be8f.redhat.com","uid":"dcc477d5-a09b-4bfe-aad7-186f36639c2f","apiVersion":"coordination.k8s.io/v1","resourceVersion":"49098"}, "reason": "LeaderElection", "message": "build-service-controller-manager-688845d6ff-k69v5_ced1347f-31b8-40a4-9e09-dd51eebc639f became leader"}
1.659127935978382e+09	INFO	Starting EventSource	{"controller": "component", "controllerGroup": "appstudio.redhat.com", "controllerKind": "Component", "source": "kind source: *v1alpha1.Component"}
1.6591279359784064e+09	INFO	Starting Controller	{"controller": "component", "controllerGroup": "appstudio.redhat.com", "controllerKind": "Component"}
1.6591279359790947e+09	INFO	Starting EventSource	{"controller": "pipelinerun", "controllerGroup": "tekton.dev", "controllerKind": "PipelineRun", "source": "kind source: *v1beta1.PipelineRun"}
1.659127935979131e+09	INFO	Starting Controller	{"controller": "pipelinerun", "controllerGroup": "tekton.dev", "controllerKind": "PipelineRun"}
1.6591279360796182e+09	INFO	Starting workers	{"controller": "pipelinerun", "controllerGroup": "tekton.dev", "controllerKind": "PipelineRun", "worker count": 1}
1.659127936079685e+09	INFO	Starting workers	{"controller": "component", "controllerGroup": "appstudio.redhat.com", "controllerKind": "Component", "worker count": 1}

```

but it seems like gitops thinks build-service is only "progressing" based on the e2e log

I'll stop here, wait to see what @mmorhun says on Monday.

just in case 

/retest

**Comment by gabemontero** [comment]
e2e's are green now

File: main.go
Comment (mmorhun) [inline main.go:85]: Could you move this functionality into a separate function?

**Review by mmorhun — APPROVED** [review]
LGTM

File: main.go
Comment (gabemontero) [reply main.go]: functionality moved to separate function @mmorhun .... PTAL / thanks

**Comment by gabemontero** [comment]
all green e2e @mmorhun minus the doesn't mark as complete pipeline as code item that appears across repos

shall we merge?  or does a second look bring some more comments to mind?

thanks

File: main.go
Comment (mmorhun) [reply main.go]: Thank you, @gabemontero. I think we can merge it.

Comment (gabemontero) [reply main.go]: ok cool - do you want to apply the lgtm label via the lgtm prow command, or should we just hit the merge button @mmorhun ?

**Pull request merged**
